Observer/OrderSaveAfter.php

```diff
@@ -49,39 +77,66 @@ function execute(Observer $ob) {
 		}
 	}
 
+	/**
+	 * @param array $payload
+	 * @return false|string
+	 */
+	private static function encode(array $payload)
+	{
+		$response = json_encode($payload, JSON_PRETTY_PRINT|JSON_UNESCAPED_SLASHES|JSON_UNESCAPED_UNICODE);
+
+		// Log errors
+		if (json_last_error() !== JSON_ERROR_NONE) {
+			self::$encoding_error_msg = 'Stock2Shop Webhook JSON encoding error: ' . json_last_error_msg();
```

Comment:
I think this being a static variable could possibly be a problem - cos it might persist from one call to the function execute(Observer $ob) to another ...

Or is that paranoid, @flumonion ?

Can we sure that these will only be called once within a particular process?

Reply:
I don't think it should be an issue, but just in case, I reset the variables to null at the top of the execute method. This should take care of it.